UI/MemStickScreen.cpp

```diff
@@ -0,0 +1,155 @@
+
+	root_ = new AnchorLayout();
+
+	ViewGroup *columns = new LinearLayout(ORIENT_HORIZONTAL, new AnchorLayoutParams(FILL_PARENT, FILL_PARENT));
+	root_->Add(columns);
+
+	ViewGroup *leftColumn = new AnchorLayout(new LinearLayoutParams(1.0));
+	ViewGroup *rightColumnItems = new LinearLayout(ORIENT_VERTICAL, new LinearLayoutParams(300, FILL_PARENT, actionMenuMargins));
+	columns->Add(leftColumn);
+	columns->Add(rightColumnItems);
+
+	Path path = g_Config.memStickDirectory;
+	int64_t freeSpaceAtMemStick = -1;
+#if PPSSPP_PLATFORM(ANDROID)
+	if (Android_IsContentUri(path.ToString())) {
+		freeSpaceAtMemStick = Android_GetFreeSpaceByContentUri(path.ToString());
+	} else {
+		freeSpaceAtMemStick = Android_GetFreeSpaceByFilePath(path.ToString());
+	}
+#endif
+
+	int leftSide = 100;
+	settingInfo_ = new SettingInfoMessage(ALIGN_CENTER | FLAG_WRAP_TEXT, new AnchorLayoutParams(dp_xres - leftSide - 40.0f, WRAP_CONTENT, leftSide, dp_yres - 80.0f - 40.0f, NONE, NONE));
+	settingInfo_->SetBottomCutoff(dp_yres - 200.0f);
+
+	root_->Add(settingInfo_);
+
+	leftColumn->Add(new TextView(iz->T("Memory Stick Storage"), ALIGN_LEFT, false, new AnchorLayoutParams(10, 10, NONE, NONE)));
+	leftColumn->Add(new TextView(iz->T("MemoryStickDescription", "Choose where your PSP memory stick data (savegames, etc) is stored"), ALIGN_LEFT, false, new AnchorLayoutParams(10, 50, NONE, NONE)));
+	leftColumn->Add(new TextView(g_Config.memStickDirectory.ToVisualString(), ALIGN_LEFT, false, new AnchorLayoutParams(10, 140, NONE, NONE)));
+
+	std::string freeSpaceText = "Free space: N/A";
+	if (freeSpaceAtMemStick >= 0) {
+		freeSpaceText = StringFromFormat("free space: %lld bytes", freeSpaceAtMemStick);
+	}
+
+	leftColumn->Add(new TextView(freeSpaceText, ALIGN_LEFT, false, new AnchorLayoutParams(10, 240, NONE, NONE)));
+
+	rightColumnItems->Add(new Choice(iz->T("Browse")))->OnClick.Handle(this, &MemStickScreen::OnBrowse);
+	rightColumnItems->Add(new Choice(di->T("Back")))->OnClick.Handle<UIScreen>(this, &UIScreen::OnOK);
+}
+
+UI::EventReturn MemStickScreen::OnBrowse(UI::EventParams &params) {
+	auto sy = GetI18NCategory("System");
+	System_SendMessage("browse_folder", "");
+	return UI::EVENT_DONE;
+}
+
+void MemStickScreen::sendMessage(const char *message, const char *value) {
+	// Always call the base class method first to handle the most common messages.
+	UIDialogScreenWithBackground::sendMessage(message, value);
+
+	if (screenManager()->topScreen() == this) {
+		if (!strcmp(message, "browse_folderSelect")) {
+			std::string filename;
+			filename = value;
+			INFO_LOG(SYSTEM, "Got folder: '%s'", filename.c_str());
+			pendingMemStickFolder_ = Path(filename);
+			CallbackMemStickFolder(true);
+		}
+	}
+}
+
+void MemStickScreen::CallbackMemStickFolder(bool yes) {
+	auto sy = GetI18NCategory("System");
+
+	if (yes) {
+		Path memStickDirFile = Path(g_Config.internalDataDirectory) / "memstick_dir.txt";
+		Path testWriteFile = Path(pendingMemStickFolder_) / ".write_verify_file";
```

Comment:
Duplicate `Path()`.  I think on `internalDataDirectory` too.

-[Unknown]

Reply:
Oops, remains of older code.